src/lqcut.jl

```diff
@@ -1,58 +1,77 @@
 """
-    lq_drift_time_correction(lq_norm::Vector{Float64}, tdrift, e_cal; DEP_left=1589u"keV", det_icpc = false, DEP_right=1596u"keV", lower_exclusion=0.005, upper_exclusion=0.98, drift_cutoff_sgima=2.0)
+    lq_drift_time_correction(lq_norm::Vector{Float64}, tdrift, e_cal; DEP_left=1589u"keV", det_icpc = false, DEP_right=1596u"keV", lower_exclusion=0.005, upper_exclusion=0.98, drift_cutoff_sigma=2.0)
 
 Perform the drift time correction on the LQ data using the DEP peak. The function cuts outliers in lq and drift time, then performs a linear fit on the remaining data. The data is Corrected by subtracting the linear fit from the lq data.
 # Returns
     * `result`: NamedTuple of the corrected lq data, the box used for the linear fit and the drift time function
     * `report`: NamedTuple of the histograms used for the fit
 """
-function lq_drift_time_correction(lq_norm::Vector{Float64}, tdrift, e_cal, detectortype; DEP_left=1589u"keV", DEP_right=1596u"keV", lower_exclusion=0.005, upper_exclusion=0.98, drift_cutoff_sgima=2.0)
+function lq_drift_time_correction(
+    lq_norm::Vector{Float64}, tdrift::Vector{<:Unitful.RealOrRealQuantity}, e_cal::Array{<:Unitful.Energy{<:Real}}, DEP_µ::Unitful.AbstractQuantity, DEP_σ::Unitful.AbstractQuantity;
+     DEP_edgesigma::Float64 = 3.0 , mode::Symbol = :gaussian, lower_exclusion::Float64 = 0.005, upper_exclusion::Float64 = 0.98, drift_cutoff_sigma::Float64 = 2.0,
+     e_expression::Union{String,Symbol}="e")
 
-    #Using fixed values for DEP, can be changed to use values from DEP fit from Energy calibration 
-    lq_DEP_dt = lq_norm[DEP_left .< e_cal .< DEP_right]
+
+    #calculate DEP edges
+    DEP_left = DEP_µ - DEP_edgesigma * DEP_σ
+    DEP_right = DEP_µ + DEP_edgesigma * DEP_σ
+
+    #cut data to DEP peak
+    lq_DEP = lq_norm[DEP_left .< e_cal .< DEP_right]
     t_tcal = ustrip.(tdrift[DEP_left .< e_cal .< DEP_right])
 
     #lq cutoff
-    #sort array to exclude outliers
-    sort_lq = sort(lq_DEP_dt)
-    low_cut = Int(round(length(sort_lq) * lower_exclusion)) # cut lower 0.5%
-    high_cut = Int(round(length(sort_lq) * upper_exclusion)) # cut upper 2%
-    lq_prehist = fit(Histogram, lq_DEP_dt, range(sort_lq[low_cut], sort_lq[high_cut], length=100))
+    #sort array to exclude outliers for better fit
+    low_cut_value = quantile(lq_DEP, lower_exclusion)  # 0.5% quantile
+    high_cut_value = quantile(lq_DEP, upper_exclusion)  # 2% quantile
+
+    ideal_bin_width = LegendSpecFits.get_friedman_diaconis_bin_width(lq_DEP[low_cut_value .< lq_DEP .< high_cut_value])
+    ideal_length = Int(round(abs(high_cut_value - low_cut_value) / ideal_bin_width))
+
+    lq_prehist = fit(Histogram, lq_DEP, range(low_cut_value, high_cut_value, length=ideal_length))
```

Comment:
In `range`: you could use `step = ideal_bin_width`, or is there any particular reason why you are defining this via `length`?